lib/travis/remote_vcs/user.rb

```diff
@@ -13,7 +13,7 @@ def auth_request(provider: :github, redirect_uri:, state:)
         end
         return JSON.parse(resp.body)['data'] if resp.success?
 
-        halt 401, 'unauthorized'
+        raise ConnectionError
```

Comment:
@weban, I would provide additional details here, status code and body and log it, otherwise we will be masking the cause and it will be harder to debug